hw/ip/aes/doc/aes.md

```diff
@@ -0,0 +1,296 @@
+    ]
+  ]
+}
+```
+
+The AES unit is configured to do decryption (`Config mode` = DECRYPT).
+Once the new key has been provided via the control and status registers (top), this new key is loaded into the Full Key register (`key_full` = K0-3) and the KEM starts performing encryption (`KEM mode`=ENCRYPT).
+The cipher data path remains idle (`AES mode`=IDLE).
+In every round, the value in `key_full` is updated.
+After 10 encryption rounds, the value in `key_full` equals the start key for decryption.
+This value is stored into the Decryption Key register (`key_dec` = K0-3’ at the very bottom).
+Now the AES unit can switch between encryption/decryption without overhead as both the start key for encryption (`key_init`) and decryption (`key_dec`) can be loaded into `full_key`.
+
+For details on the KeyExpand operation refer to the [AES specification, Section 5.2](https://csrc.nist.gov/csrc/media/publications/fips/197/final/documents/fips-197.pdf).
+
+Key expanding is the only operation in the AES unit for which the functionality depends on the selected key length.
+Having a KEM that supports 128-bit key expansion, support for the 256-bit mode can be added at low overhead.
+In contrast, the 192-bit mode requires much larger muxes.
+Support for this mode is thus optional and can be enabled/disabled via a design-time parameter.
+
+Once we have cost estimates in terms of gate count increase for 192-bit mode, we can decide on whether or not to use it in OpenTitan.
+Typically, systems requiring security above AES-128 go directly for AES-256.
+
+
+{{% section2 Hardware Interfaces }}
+
+{{% hwcfg Name }}
+
+
+{{% section2 Design Details }}
+
+This section discusses different design details of the AES module.
+
+
+### System Key-Manager Interface
+
+This version of the AES unit is controlled entirely by the processor.
+The processor writes both input data as well as the initial key to dedicated registers via the system bus interconnect.
+
+Future versions of the AES unit might include a separate interface through which a possible system key manager can update the initial key without exposing the key to the processor or other hosts attached to the system bus interconnect.
+
+
+### Security Hardening
+
+The primary focus of the first version of the AES unit lies in having a first functional implementation.
+
+Future efforts on this unit will have a focus on security hardening of the design.
+Future versions of this AES unit thus might employ different means at architectural, microarchitectural and physical levels to reduce information leakage (e.g. power and electromagnetic) and aggravate potential side-channel attacks.
+
+
+{{% section1 Programmers Guide }}
+
+This section discusses how software can interface with the AES unit.
+
+
+{{% section2 Initialization }}
+
+The initialize the AES unit, software must write the initial key to the Initial Key register.
+Independent of the selected key length, software must always write all 8 32-bit registers.
```

Comment:
Curious, why do all 8 registers need to be written in 128b case?

Reply:
The AES unit needs to know when to start the generation of the decryption key. This should happen only after the full initial key has been provided. I could change this to take into account the current value of the key length config register. I did not to this so far to ease DV.